### PR TITLE
Moving sensor-ready stream to synchronizer.

### PR DESCRIPTION
**Pull request by sukritkalra**

### CODE REVIEW

File: data_gatherer.py [hunk 1]
```diff
@@ -45,7 +45,8 @@ def main(argv):
                                       pylot.utils.Rotation())
 
     control_loop_stream = erdos.LoopStream()
-    notify_stream = erdos.LoopStream()
+    release_sensor_stream = erdos.IngestStream()
+    notify_streams = []
```
Comment (ICGog) [inline data_gatherer.py:49]: Please remove notify_streams. They're not needed anymore.

File: pylot.py [hunk 2]
```diff
@@ -165,7 +178,13 @@ def driver():
                                                   point_cloud_stream,
                                                   ground_segmented_stream,
                                                   imu_stream, pose_stream)
-    erdos.run()
+    erdos.run_async()
+
+    # If we did not use the pseudo-asynchronous mode, ask the sensors to
```
Comment (ICGog) [inline pylot.py:183]: Is this needed? I don't think simulator_in_sync is used in any non pseudo-asynchronous mode.

File: scripts/compute_decay.py [hunk 3]
```diff
@@ -48,25 +48,31 @@ def main(argv):
                                       pylot.utils.Rotation())
 
     control_loop_stream = erdos.LoopStream()
-    notify_stream1 = erdos.LoopStream()
-    notify_stream2 = erdos.LoopStream()
+    release_sensor_stream = erdos.IngestStream()
+    notify_streams = []
```
Comment (ICGog) [inline scripts/compute_decay.py:52]: As above.

File: pylot.py (hunk 2 above)
Comment (sukritkalra) [reply pylot.py:183]: I think so. In non pseudo-asynchronous mode, we would not have `PlanningPoseSynchronizerOperator` and thus, noone to tell the sensor streams to release data. This line just forces the sensors to release all the data as soon as it is available.

**Pull request merged**